app/src/main/java/cm/aptoide/pt/home/BottomNavigationNavigator.java

```diff
@@ -0,0 +1,161 @@
+  private final FragmentNavigator fragmentNavigator;
+  private final String defaultStoreName;
+  private final BottomNavigationAnalytics bottomNavigationAnalytics;
+  private final SearchAnalytics searchAnalytics;
+  private final int homePosition = 0;
+  private final int searchPosition = 1;
+  private final int storesPosition = 2;
+  private final int appsPosition = 3;
+  private ArrayList<Integer> bottomNavigationItems;
+
+  public BottomNavigationNavigator(FragmentNavigator fragmentNavigator, String defaultStoreName,
+      BottomNavigationAnalytics bottomNavigationAnalytics, SearchAnalytics searchAnalytics) {
+    this.defaultStoreName = defaultStoreName;
+    this.bottomNavigationAnalytics = bottomNavigationAnalytics;
+    this.searchAnalytics = searchAnalytics;
+    bottomNavigationItems = new ArrayList<>();
+    this.fragmentNavigator = fragmentNavigator;
+  }
+
+  public void navigateToBottomNavigationItem(int bottomNavigationPosition) {
+    switch (bottomNavigationPosition) {
+      case homePosition:
+        bottomNavigationAnalytics.sendNavigateToHomeClickEvent();
+        BottomHomeFragment bottomHomeFragment = new BottomHomeFragment();
+        navigateToHome(bottomHomeFragment);
+        break;
+      case searchPosition:
+        bottomNavigationAnalytics.sendNavigateToSearchClickEvent();
+        searchAnalytics.searchStart(SearchSource.BOTTOM_NAVIGATION, true);
+        SearchResultFragment searchResultFragment =
+            SearchResultFragment.newInstance(defaultStoreName, true);
+        navigateToSearch(searchResultFragment);
+        break;
+      case storesPosition:
+        bottomNavigationAnalytics.sendNavigateToStoresClickEvent();
+        MyStoresFragment myStoresFragment =
+            MyStoresFragment.newInstance(getStoreEvent(), "default", "stores", StoreContext.home);
+        navigateToStore(myStoresFragment);
+        break;
+      case appsPosition:
+        bottomNavigationAnalytics.sendNavigateToAppsClickEvent();
+        AppsFragment appsFragment = new AppsFragment();
+        navigateToApps(appsFragment);
+        break;
+    }
+  }
+
+  public void navigateToHome(BottomHomeFragment bottomHomeFragment) {
+    navigateToSelectedFragment(0, bottomHomeFragment);
+  }
+
+  public void navigateToSearch(SearchResultFragment searchResultFragment) {
+    navigateToSelectedFragment(1, searchResultFragment);
+  }
+
+  public void navigateToStore(MyStoresFragment myStoresFragment) {
+    navigateToSelectedFragment(2, myStoresFragment);
+  }
+
+  public void navigateToApps(AppsFragment appsFragment) {
+    navigateToSelectedFragment(3, appsFragment);
+  }
+
+  public void navigateBack() {
+    int bottomNavigationPosition = bottomNavigationItems.get(1);
+    bottomNavigationItems.remove(0);
+    navigateBackToBottomNavigationItem(bottomNavigationPosition);
+  }
+
+  public boolean canNavigateBack() {
+    return bottomNavigationItems.size() > 1;
+  }
+
+  private void checkAndReplaceItem(int newItem) {
```

Comment:
Could use bottomNavigationItems.indexOf to find the listPosition variable that is being calculated here